project-base/tests/ReadModelBundle/Functional/Product/Listed/ListedProductViewRepositoryTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Tests\ReadModelBundle\Functional\Product\Listed;
+
+use Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData;
+use Shopsys\FrameworkBundle\Model\Product\Listing\ProductListOrderingConfig;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductViewInterface;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductViewRepository;
+use Tests\ShopBundle\Test\FunctionalTestCase;
+
+class ListedProductViewRepositoryTest extends FunctionalTestCase
+{
+    public function testGetForAccessories(): void
+    {
+        /** @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewRepository $listedProductViewRepository */
+        $listedProductViewRepository = $this->getContainer()->get(ListedProductViewRepository::class);
+
+        $listedProductViews = $listedProductViewRepository->getForAccessories(1);
+
+        $this->assertCount(2, $listedProductViews);
+
+        $this->assertArrayHasKey(24, $listedProductViews);
+        $this->assertArrayHasKey(13, $listedProductViews);
+
+        $this->assertInstanceOf(ListedProductViewInterface::class, $listedProductViews[24]);
+        $this->assertInstanceOf(ListedProductViewInterface::class, $listedProductViews[13]);
+
+        $this->assertEquals('Kabel HDMI A - HDMI A M/M 2m gold-plated connectors High Speed HD', $listedProductViews[24]->getName());
+        $this->assertEquals('Defender 2.0 SPK-480', $listedProductViews[13]->getName());
+    }
+
+    public function testGetPaginatedForBrand(): void
+    {
+        /** @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewRepository $listedProductViewRepository */
+        $listedProductViewRepository = $this->getContainer()->get(ListedProductViewRepository::class);
+
+        $paginationResults = $listedProductViewRepository->getPaginatedForBrand(1, ProductListOrderingConfig::ORDER_BY_NAME_ASC, 1, 10);
```

Comment:
If you'll use the "persistent reference" for product, use it also for the brand

Reply:
I made this hardcoded values into variables for better understanding.